frontend/apps/filemanager/filemanager.lua

```diff
@@ -157,6 +157,30 @@ function FileManager:init()
                         UIManager:close(self.file_dialog)
                     end,
                 },
+                {
+                    text = _("Purge .sdr"),
+                    enabled = lfs.attributes(util.realpath(file), "mode") == "file" and util.realpath(file):match("%.") and
+                            lfs.attributes(util.realpath(file):match("(.*)%.")..".sdr", "mode") == "directory",
+                    callback = function()
+                        local purgeDir = require("ffi/util").purgeDir
+                        local full_path = util.realpath(file)
+                        local sidecar = full_path:match("%.") and full_path:match("(.*)%.")..".sdr" or nil
```

Comment:
If you assume there is another process operating the filesystem, then it's possible for that file to go away any time, including the moment right after we do the `lfs` check and before we act on the sidecard directory. The best we can do here is to make sure we don't have race condition without koreader's code unfortunately ;)

Then again, you can ignore my comment here since this code will be replaced by the new hasSidecarDir function.